tests/cxx/typeid.cc

```diff
@@ -0,0 +1,37 @@
+//===--- typeid.cc - test input file for iwyu -----------------------------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+// IWYU_ARGS: -I .
+
+// Tests that IWYU suggests an include instead of a forward declaration for
+// expressions used as arguments to typeid(). This is necessary as calling
+// typeid() on an incomplete type causes the program to be ill-formed.
+
+#include "tests/cxx/direct.h"
+#include <typeinfo>
```

Comment:
It would be nice not to have to pull this standard library header in. Is it possible to do `(void)typeid(indirect);` and still see the full use of `IndirectClass`?

Reply:
> The type std​::​type_­info ([[type.info]](https://eel.is/c++draft/type.info)) is not predefined; if a standard library declaration ([[typeinfo.syn]](https://eel.is/c++draft/typeinfo.syn), [[std.modules]](https://eel.is/c++draft/std.modules)) of std​::​type_­info does not precede ([[basic.lookup.general]](https://eel.is/c++draft/basic.lookup.general)) a typeid expression, the program is ill-formed[.](https://eel.is/c++draft/expr.typeid#7.sentence-1)

https://eel.is/c++draft/expr.typeid

Reply:
:eyes: Oh, C++, you leave me no comfort. @bolshakov-a Thanks for the heads-up!